daemon/logger/journald/read.go

```diff
@@ -392,6 +396,76 @@ func (s *journald) readLogs(logWatcher *logger.LogWatcher, config logger.ReadCon
 	return
 }
 
+func (s *journald) readRemoteLogs(logWatcher *logger.LogWatcher, config logger.ReadConfig) {
+	client := http.Client{}
+
+	// Build a url to get logs from systemd-journal-gatewayd.
+	container_id := s.vars["CONTAINER_ID_FULL"]
+	v := url.Values{}
+	v.Set("CONTAINER_ID_FULL", container_id)
+	if config.Follow {
+		v.Set("follow", "")
+	}
+	u := url.URL{
+		Scheme:   "http",
+		Host:     s.gatewayAddress,
+		Path:     "entries",
+		RawQuery: v.Encode(),
+	}
+
+	// Send a request to get logs and get the response from systemd-journal-gatewayd.
+	req, err := http.NewRequest("GET", u.String(), nil)
+	req.Header.Add("Accept", "application/json")
+	resp, err := client.Do(req)
+
+	defer close(logWatcher.Msg)
+	if resp != nil && resp.Body != nil {
+		defer resp.Body.Close()
+	}
+
+	// Handle errors while opening journal.
+	fmtString := "error %q while opening journal from %s for container %q"
+	if err != nil {
+		logWatcher.Err <- fmt.Errorf(fmtString, err, s.gatewayAddress, container_id)
+		return
+	} else if resp != nil && resp.StatusCode >= 400 {
```

Comment:
It will be much more go-ish if extra `else` is removed:
https://golang.org/doc/effective_go.html#if